File: python/leetcode27.py
class Solution:
    def removeElement(self, nums: list[int], val: int) -> int:
        n = len(nums)
        l = 0
        for r in range(0, n):
            if nums[r] != val:
                nums[l] = nums[r]
                l += 1
        return l


solution = Solution()
lst = [0, 1, 2, 2, 3, 0, 4, 2]
print(solution.removeElement(lst, 2))

File: C++/leetcode27.cpp
#include<iostream>
#include<vector>
using namespace std;
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
    //双指针右指针 
    // right
    // right 指向当前将要处理的元素，左指针 
    // left
    // left 指向下一个将要赋值的位置。
        int l=0,n = nums.size();
        for(int r=0;r<n;r++){
            if(nums[r]!=val){
                nums[l]=nums[r];
                l++;
            }
        }
        return l;
    }
};